### ISSUE TITLE
Encountering an Out of Memory issue that appears to be related to method type resolution.

### ISSUE THREAD

**Issue by tkvangorder** [openrewrite/rewrite#1312]
```
typeSignatureParamTypes.size() > typeDescriptorParamTypes.size() for method com.google.common.collect.AbstractMapBasedMultimap$WrappedSortedSet.<init>
io.github.classgraph.ClassGraphException: typeSignatureParamTypes.size() > typeDescriptorParamTypes.size() for method com.google.common.collect.AbstractMapBasedMultimap$WrappedSortedSet.<init>
        at io.github.classgraph.MethodInfo.getParameterInfo(MethodInfo.java:496)
        at org.openrewrite.java.internal.ClassgraphJavaTypeSignatureBuilder.methodSignature(ClassgraphJavaTypeSignatureBuilder.java:247)
        at org.openrewrite.java.internal.ClassgraphTypeMapping.methodType(ClassgraphTypeMapping.java:227)
        at org.openrewrite.java.internal.ClassgraphTypeMapping.type(ClassgraphTypeMapping.java:140)
        at org.openrewrite.java.internal.ClassgraphTypeMapping.type(ClassgraphTypeMapping.java:85)
        at org.openrewrite.java.marker.JavaSourceSet.build(JavaSourceSet.java:70)
        at org.openrewrite.gradle.GradleProjectParser.parse(GradleProjectParser.java:236)
        at org.openrewrite.gradle.GradleProjectParser.parse(GradleProjectParser.java:151)
        at org.openrewrite.gradle.GradleProjectParser.listResults(GradleProjectParser.java:278)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.openrewrite.gradle.DelegatingProjectParser.lambda$listResults$5(DelegatingProjectParser.java:101)
        at org.openrewrite.gradle.DelegatingProjectParser.unwrapInvocationException(DelegatingProjectParser.java:116)
        at org.openrewrite.gradle.DelegatingProjectParser.listResults(DelegatingProjectParser.java:101)
        at org.openrewrite.gradle.AbstractRewriteTask.listResults(AbstractRewriteTask.java:47)
        at org.openrewrite.gradle.RewriteRunTask.run(RewriteRunTask.java:55)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
```

https://github.com/classgraph/classgraph/blob/874442f4a1c76b2bd63284988d20c8776e0b0310/src/main/java/io/github/classgraph/MethodInfo.java#L497

Eventually exits with a JVM OOM error:

```
Expiring Daemon because JVM heap space is exhausted
Exception in thread "Connection evictor" java.lang.OutOfMemoryError: Java heap space
Exception in thread "Daemon periodic checks" java.lang.OutOfMemoryError: Java heap space
MethodInfo.java
                        "typeSignatureParamTypes.size() > typeDescriptorParamTypes.size() for method "
```

**Comment by traceyyoshima** [comment]
It looks like the key compression fixed the OOM. The `typeSignatureParamTypes.size() > typeDescriptorParamTypes.size()` exception is thrown from Classgraph and may be looked into separately.

**Comment by lukehutch** [comment]
The `typeSignatureParamTypes.size() > typeDescriptorParamTypes.size()` bug should be able to be fixed by upgrading to ClassGraph-4.8.143.
